src/Services/CandlesHistoryServiceProvider.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using Core.Domain.Market;
+using Core.Services;
+using Lykke.Service.CandlesHistory.Client;
+
+namespace Services
+{
+    public class CandlesHistoryServiceProvider : ICandlesHistoryServiceProvider
+    {
+        private readonly Dictionary<MarketType, ICandleshistoryservice> _services;
+
+        public CandlesHistoryServiceProvider()
+        {
+            _services = new Dictionary<MarketType, ICandleshistoryservice>();
+        }
+
+        public void RegisterMarket(MarketType marketType, string connectionString)
+        {
+            _services.Add(marketType, new Candleshistoryservice(new Uri(connectionString)));
+        }
+
+        public ICandleshistoryservice TryGet(MarketType market)
```

Comment:
Will throw exception if market is not set, but name and usage tells us, that null is expected in such case.